File: learn_android/code/JAVA/ViewModelSavedStatetest1/app/src/main/java/com/hkc/viewmodelsavedstatetest1/MyViewModel.java
package com.hkc.viewmodelsavedstatetest1;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.SavedStateHandle;
import androidx.lifecycle.ViewModel;

public class MyViewModel extends ViewModel {
//    MutableLiveData<Integer> number;
    private SavedStateHandle handle;

    public MyViewModel(SavedStateHandle handle){
        this.handle = handle;
    }

    public MutableLiveData<Integer> getNumber() {
        if(!handle.contains(MainActivity.KEY_NUMBER)){
            handle.set(MainActivity.KEY_NUMBER,0);
        }
        return handle.getLiveData(MainActivity.KEY_NUMBER);
//        if(number == null){
//            MutableLiveData<Integer> number = new MutableLiveData<>();
//            number.setValue(0);
//        }
//        return number;
    }
    public void addNumber(){
        getNumber().setValue(getNumber().getValue() +1);
//        number.setValue(number.getValue()+1);
    }


}

File: learn_android/code/GitDemo1/app/src/main/java/com/hkc/gitdemo1/MyViewModel.kt
package com.hkc.gitdemo1

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MyViewModel:ViewModel() {
    private val _numberLiveData =  MutableLiveData(0)
    val numberLiveData : LiveData<Int> get() = _numberLiveData

    fun add(n:Int){
        _numberLiveData.value = _numberLiveData.value?.plus(n)

    }
    fun reset(){
        _numberLiveData.value = 0
    }

//    var number = 0
}